### DISCUSSION TITLE
Integrate a paper backup of SSH key/passphrase

### DISCUSSION THREAD

**Discussion by jose1711** [borgbase/vorta#986] [Ideas]
Borg includes a way to print repository key on a sheet of paper (optionally as a QR code). At this moment `vorta` does not provide a way to run such export from its interface. Most importantly though a backup of SSH key and passphrase should be included with such export because usage of `keyfile` is limited (and often may not be needed). Output then should be sent to printer and stored in a safe place to have a way to access the backup if the client is compromised.

Maybe something like this (extra button on `Repository` tab):

Create paper backup for `profile` including:
 - [x] repository URL (mandatory - always checked)
 - [ ] passphrase
 - [ ] repository key (owned by repository/this host)
 - [ ] SSH key
 - [ ] custom data (could be ssh key passphrase or hint), specify here: `......`

**Comment by jose1711** [comment]
Very simple implementation in shell that works for me but probably won't work for anyone else: https://github.com/jose1711/vorta2paper/blob/master/vorta2paper.sh but at least gives some idea.

On my system it results in:
![obrázok](https://user-images.githubusercontent.com/1406222/118884055-4429fc80-b8f6-11eb-8538-6089196c6bff.png)

**Comment by goebbe** [comment]
Nice script! Thank you for sharing! 

This script should/ could be extended to add the keyfile of the repository (local or repokey). Without the borg keyfile of your repository you will not be able decrypt the backup. 
If you use the repokey (default in Vorta) the keyfile is stored with the repo on the backup-server and the repo passphrase would be sufficient for recovery. 
Anyway borg recommends to make a backup of the keyfile when initialising the repository - just in case the repokey gets corrupted. 

I believe the paper-backup of the SSH-keys is more about convenience, since you could always set up new SSH-keys (even from a different computer), provided you have access to your backup-server. 

edit: I just realized, that you already point to borg keyfiles in the first post.
